src/features/access-package/edit/client_selection.jsx

```diff
@@ -0,0 +1,140 @@
+import {makeStyles} from "@material-ui/core/styles";
+import {useDispatch, useSelector} from "react-redux";
+import {
+    Avatar,
+    List,
+    ListItem,
+    ListItemAvatar,
+    ListItemSecondaryAction,
+    ListItemText,
+    Typography
+} from "@material-ui/core";
+import ClientIcon from "@material-ui/icons/ImportantDevices";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Switch from "@material-ui/core/Switch";
+import FormControl from "@material-ui/core/FormControl";
+import {updateAccessPackages} from "../../../data/redux/actions/access_package";
+import FeatureHelperText from "../../../common/help/FeatureHelperText";
+import WarningMessageBox from "../../../common/message-box/WarningMessageBox";
+
+const useStyles = makeStyles(theme => ({
+    root: {
+        marginTop: theme.spacing(2),
+        marginLeft: theme.spacing(3),
+        marginRight: theme.spacing(3),
+    },
+    clientList: {
+        maxWidth: 800,
+        margin: "auto",
+    },
+    itemAvatar: {
+        color: "#fff",
+        backgroundColor: theme.palette.secondary.main
+    },
+    header: {
+        marginTop: theme.spacing(4),
+        marginLeft: theme.spacing(2),
+        marginBottom: theme.spacing(1),
+    },
+}));
+
+const ClientSelection = (props) => {
+    const classes = useStyles();
+    const {selectedAccessPackage} = props;
+    const clients = useSelector(state => state.client.clients);
+    const accessPackages = useSelector(state => state.access_package.accessPackages);
+    const [showWarning, setShowWarning] = useState(false);
+    const [switchValue, setSwitchValue] = useState(false);
+    const [switchClient, setSwitchClient] = useState(null);
+    const warningMessageText = "Når du aktiverer en aksesspakke på en klient, vil andre aksesspakker som er koblet til denne klienten bli fjernet. Ønsker du å fortsette?";
+
+
+    const dispatch = useDispatch();
+
+    function findIndex(array, value) {
+        for (let i = 0; i < array.length; i += 1) {
+            if (array[i].dn === value.dn) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    function clientChange(doChange) {
+        if (doChange) {
+            handleClientChange(switchValue, switchClient);
+        }
+        setShowWarning(false);
+    }
+
+    function handleClientChange(event, client) {
+        let newClients = [...selectedAccessPackage.clients];
+        let newAccessPackages = [...accessPackages];
+        let newAccessPackage = {...selectedAccessPackage};
+        const accessPackageIndex = findIndex(newAccessPackages, newAccessPackage);
+        if (newClients.includes(client.dn)) {
+            let clientIndex = newClients.indexOf(client.dn);
+            newClients.splice(clientIndex, 1);
+        } else {
+            newClients.push(client.dn);
+        }
+        newAccessPackage.clients = newClients;
+        newAccessPackages[accessPackageIndex] = newAccessPackage;
```

Comment:
can possibly be -1 and give an out of index...

Reply:
Fixed by using array indexOf and filters. removed method